### DISCUSSION TITLE
Custom Dispatcher

### DISCUSSION THREAD

**Discussion by rohank-dev** [AbsaOSS/spline-spark-agent#681] [Q&A] [answered]
Hello All,

I am planning to add a custom dispatcher, which is in line with existing kafka dispatcher, where I would like to add some checks before sending the data. 
As per the link https://github.com/AbsaOSS/spline-spark-agent#creating-your-own-dispatcher, I am able to create a custom dispatcher, however my question is will my custom dispatcher will take precedence over KafkaLineageDispatcher or will it be causing conflict. Since both will be enabled to use Kafka as a dispatcher.

Please suggest.

Thanks,
Rohan

**Comment by cerveada** [comment, marked as answer]
Hi, 

- if you want to use just your dispatcher define `spline.lineageDispatcher=my-dispatcher` and nothing else will be used.
- if you want to use multiple dispatchers you must define one root dispatcher that will be calling the others, this can be either Composite Dispatcher or Fallback Dispatcher.

See:
https://github.com/AbsaOSS/spline-spark-agent#using-the-fallback-dispatcher
https://github.com/AbsaOSS/spline-spark-agent#using-the-composite-dispatcher

**Reply by wajda** [reply]
Just use common sense :)
